### ISSUE TITLE
Multiple warnings about deprecated preprocessor declarations

### ISSUE THREAD

**Issue by kozross** [philburk/pforth#40]
These all look similar to this:

```
/usr/include/features.h:183:3: warning: #warning "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE" [-Wcpp]
 # warning "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE"
```

**Comment by philburk** [comment]
I suspect this may be related to this patch from another developer. 

  https://github.com/philburk/pforth/pull/7

I am open to advice on the proper way to resolve this.

**Comment by philburk** [comment]
@letoh - can you take a look at this? It may be related to your pForth patch.

If we use _DEFAULT_SOURCE then it does compile on older systems.
If we use _GNU_SOURCE then we get deprecation warnings.

**Comment by letoh** [comment]
@philburk I guess that it will missing some default definitions on the older system if we just simply remove _DEFAULT_SOURCE (though I tested on the glibc 2.17 environment without any warning).

I just reviewed the original issue to find out the root cause. On my old platform, the ECHOCTL flag is declared only when _GNU_SOURCE (or _BSD_SOURCE) is defined. So I suggest add _DEFAULT_SOURCE back, and keep _GNU_SOURCE for the system which lacks ECHOCTL flag. I'll create a new pull request soon and sorry for inadequate patch.
